compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ReachabilityAnalyzer.java

```diff
@@ -988,6 +1007,8 @@ public static class AnalyzerData {
         boolean failureHandled;
         boolean returnedWithinQuery;
         boolean skipFurtherAnalysisInUnreachableBlock;
+        boolean hasPanicStatement;
```

Comment:
How is this different to errorThrown?

Reply:
Can't we use the same? We seem to have multiple fields and I'm wondering if we can probably clean these up.

Reply:
`errorThrown` had been used for on fail clause related scenarios. Will check whether we can clean these up. Created https://github.com/ballerina-platform/ballerina-lang/issues/36951 to track this.